### ISSUE TITLE
Speed up jail creation

### ISSUE THREAD

**Issue by xerovis** [iocage/iocage#247]
I am currently testing jail creation speeds. I wrote a script that creates 100 jails to test this. They are extremely simple jails "iocage create tag=$i"

The script started at 13:10:08 and finished at 13:17:25, 437 seconds or approx 4.37 seconds per jail. I am not complaining about how long it takes to create a jail, just looking for advice on any tweaks I might be able to do to speed up jail creation. The server I am running has 96GB RAM and 12 processors with nothing other than iocage on the box.

**Comment by skarekrow** [comment]
Others are most welcome to chime in here, but from my understanding ZFS is now the bottleneck for us. We tried to maximize as much speed as we could with regards to setting properties only once on creation, etc. I cannot think of any other tweak you can do. Maybe someone else will have one :)

@pannon any ideas?

**Comment by xerovis** [comment]
Thanks @skarekrow

**Comment by xerovis** [comment]
I ran a second test, creating 100 jails "iocage create tag=$i start=1". The script started at 13:26:43 and finished at 13:39:03 or approx 7 seconds per jail.

**Comment by skarekrow** [comment]
Odd that it's almost 60% slower. That still makes me believe it's ZFS.

**Comment by skarekrow** [comment]
I did this for fun as well on my SSD and I get a result of 1:14. So about 1.14 seconds per jail.

**Comment by xerovis** [comment]
I am running this on a Dell R710 with 15K SAS disks in RAID 1 (PERC no JBOD) so its going to be slower than your SSD :)

**Comment by skarekrow** [comment]
I figured :) I was posting for posterity as if the script was the problem I'd (hope) that our times would have been aligned. I'm hoping someone knows some magic ZFS sauce to speed these operations up. But I don't know of any myself.

**Comment by skarekrow** [comment]
If you want any more opinions on this, feel free to post it in the Google Groups. It's been about a month with no responses :)

**Comment by xerovis** [comment]
Just wondering what happened to the work going on here - https://github.com/iocage/iocage/issues/238. Did anything come of it? If not, our guys are going to do something about it as we are seeing a significant slowdown with more than 25 jails on mirrored SSDs.

I think etcd is the main candidate at the moment. Thoughts or opinions on other alternatives would be appreciated so we do not go down a path that others can see issue with.

**Comment by xerovis** [comment]
Other than reducing the number of dependencies, is there any reason why iocage does not use a distributed key/value store to store properties instead of storing them in ZFS? I can see the benefit of storing some properties in ZFS as it allows you to store/transfer basic jail configuration parameters between Hosts. But as we have expanded our wrapper scripts we have noticed there are a large number of properties stored in ZFS that are not required to be stored OR other properties for jails that would be nice to store - we have solved this by storing them in a text file attached to the jail.
